### PR TITLE
improve BaseRepository and BaseEntityService to correctly support update function

### PR DESCRIPTION
**Pull request by YuAvillachLab**
passed all test cases

### CODE REVIEW

**Comment by YuAvillachLab** [comment]
Check this:
`[INFO] Reactor Summary:
[INFO] 
[INFO] pic-sure-api ....................................... SUCCESS [  0.848 s]
[INFO] pic-sure-util ...................................... SUCCESS [  3.835 s]
[INFO] pic-sure-api-data .................................. SUCCESS [  2.003 s]
[INFO] pic-sure-resources ................................. SUCCESS [  0.025 s]
[INFO] pic-sure-resource-api .............................. SUCCESS [  3.696 s]
[INFO] PIC-SURE API war ................................... SUCCESS [  6.278 s]
[INFO] pic-sure-ga4gh-dos ................................. SUCCESS [  0.577 s]
[INFO] pic-sure-irct-resource ............................. SUCCESS [  0.462 s]
[INFO] pic-sure-aggregate-resource ........................ SUCCESS [  0.669 s]
[INFO] pic-sure-hsapi-resource ............................ SUCCESS [  0.667 s]
[INFO] pic-sure-gnome-i2b2-count-resource ................. SUCCESS [  1.028 s]
[INFO] pic-sure-api-wildfly ............................... SUCCESS [ 40.052 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:00 min
[INFO] Finished at: 2018-10-18T10:46:12-04:00
[INFO] Final Memory: 40M/135M
`

**Comment by YuAvillachLab** [comment]
image

File: pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureBaseEntityService.java [hunk 1]
```diff
@@ -102,6 +104,7 @@ public Response updateEntity(List<T> entities, BaseRepository baseRepository){
         for (T t : entities){
             boolean dbContacted = false;
             if (forAdd) {
+                t.setUuid(null);
                 baseRepository.persist(t);
```
Comment (calypsomatic) [inline pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureBaseEntityService.java]: Just curious - why do you need to set the UUID to null?

Comment (YuAvillachLab) [reply pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureBaseEntityService.java]: My thought is to provide some fault tolerance. Cuz users may not only uses this through a UI and they may copy paste user with accidentally put the uuid in it, in that case, I think maybe I should let them in as well cuz they have all other valid information

**Pull request merged**